### PR TITLE
Add restrictions to protect the owners and change their roles

### LINKED ISSUE #2089: Team Owner/Organizer issue
If you change the role of the team owner for example to Admin, it isn't possible to select it back to Owner. 
Screen Shot 2022-03-08 at 14 02 45

Also, there is an issue where the organizer of the team isn't shown as the Organizer in the confirmation email, it shows the member instead, even though the owner of the team did the confirmation.

**Case for context**
download
Screen Shot 2022-dd03-03 at 18 48 08

The organizer should be always the team owner unless the booking is confirmed by the admin within the app. Speaking of which, bookings can be confirmed by everyone in the team right now within the app, even the team members. I reckon only Admin and Owner should be able to confirm meetings.

### PR DESCRIPTION
**Pull request by juliandsg99**
## What does this PR do?

Adds restrictions so that an Owner can downgrade his role only if he isn't the only Owner. Additionally, an Owner can only modify the roles of other members who aren't Owners. Admins can only modify members with the same or lesser roles. It was considered that only an Owner can appoint another owner.

Fixes #2089 

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- The changes were tested in a team with at least two members.

## Checklist

- [X] I have read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code and corrected any misspellings
- [X] I have checked if my changes generate no new warnings

### CODE REVIEW

**Review by pumfleet — DISMISSED** [review]
You now cannot downgrade someone if you set them as an owner, which is not functionality we want

**Comment by juliansg022** [comment]
> You now cannot downgrade someone if you set them as an owner, which is not functionality we want

I understand. Currently, the Admins can downgrade other members, i.e., the Admins can downgrade members with a role equal or less to his role. 

Now, should the same rule apply to owners? the Owner should be able to downgrade members with a role equal or less to his role.

**Comment by baileypumfleet** [comment]
Yeah that's correct

**Comment by baileypumfleet** [comment]
Thank you very much, looking forward to it :)

**Review by zomars — CHANGES_REQUESTED** [review]
I've tried to update this PR but I don't have permissions. Would you kindly update your branch and solve merge conflicts?

**Comment by juliansg022** [comment]
@zomars Sure, I'll update the PR.

**Comment by juliansg022** [comment]
Hi @baileypumfleet and @zomars, I've updated the PR.

**Comment by zomars** [comment]
@juliansg022 Can I get permissions to push to this branch?

**Review by zomars — CHANGES_REQUESTED** [review]
This PR is getting little bit hard to review. We don't have permissions to keep this branch up to date. Additionally all tests are failing and type checks as well.

**Comment by juliansg022** [comment]
@zomars I updated the PR and I fixed the problem related to data types. Now the tests are working well. Regarding the permissions of the branch, this belongs to an organization so I don't have permission to give you control over it. 

I'll be updating the PR daily so you can eventually review it.

**Comment by zomars** [comment]
@juliansg022 please notify your organization owner about [our contributing guidelines](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) so we can speed up the reviewing process.

File: apps/web/components/team/MemberListItem.tsx [hunk 1]
```diff
@@ -24,6 +24,7 @@ import ConfirmationDialogContent from "@components/dialog/ConfirmationDialogCont
 import Avatar from "@components/ui/Avatar";
 import ModalContainer from "@components/ui/ModalContainer";
 
+import { useMeQuery } from "../Shell";
```
Comment (zomars) [inline apps/web/components/team/MemberListItem.tsx:27]: Is this is going to be used on another file, consider splitting it into a separate file so we can avoid circular dependency issues.

**Comment by juliansg022** [comment]
Hi @zomars and @baileypumfleet, you have been added as externals collaborators because the option to allow maintainers isn't available for organizations. https://github.community/t/how-can-we-enable-allow-edits-from-maintainers-by-default/2847/11

> @juliansg022 please notify your organization owner about [our contributing guidelines](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) so we can speed up the reviewing process.

**Review by zomars — APPROVED** [review]

**Pull request merged**